### PR TITLE
Laboratory work #3, Ulyana Abramova - 22FPL1

### PR DESCRIPTION
**Pull request by ulyanaabr**

### CODE REVIEW

File: lab_3_generate_by_ngrams/main.py [hunk 1]
```diff
@@ -45,24 +44,30 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
-        if not isinstance(text, str) or len(text) == 0:
+        if not isinstance(text, str):
+            return None
+        if not len(text) == 0:
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:49]: simplify

File: lab_3_generate_by_ngrams/main.py [hunk 2]
```diff
@@ -224,12 +235,14 @@ def _decode(self, corpus: tuple[int, ...]) -> Optional[tuple[str, ...]]:
...
-        if not isinstance(corpus, tuple) or len(corpus) == 0:
+        if not isinstance(corpus, tuple):
+            return None
+        if len(corpus) == 0:
             return None
 
         decoded_corpus = []
-        for ident in corpus:
-            token = self.get_token(ident)
+        for id in corpus:
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:244]: bad name

File: lab_3_generate_by_ngrams/main.py [hunk 3]
```diff
@@ -251,15 +264,18 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
 
         In case of corrupt input arguments, None is returned
         """
-        if not isinstance(decoded_corpus, tuple) or len(decoded_corpus) == 0:
+        if not isinstance(decoded_corpus, tuple):
+            return None
+        if not len(decoded_corpus) == 0:
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:269]: bad condition

File: lab_3_generate_by_ngrams/main.py [hunk 4]
```diff
@@ -251,15 +264,18 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
...
-
+        decoded_text = ''.join(decoded_corpus)
+        decoded_text = decoded_text.replace('_', ' ')
+        decoded_text = decoded_text.capitalize()
         if decoded_text[-1] == ' ':
-            return f"{decoded_text[:-1]}."
+            decoded_text = decoded_text[:-1]
 
-        return f"{decoded_text}."
+        return decoded_text + '.'
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:278]: fstring

File: lab_3_generate_by_ngrams/main.py [hunk 5]
```diff
@@ -686,12 +695,7 @@ def __init__(self, json_path: str, eow_token: str) -> None:
             json_path (str): Local path to assets file
             eow_token (str): Special token for text processor
         """
-        self._json_path = json_path
-        self._eow_token = eow_token
-        with open(json_path, 'r', encoding="utf-8") as file:
-            self._content = json.load(file)
-        self._text_processor = TextProcessor(self._eow_token)
-        self._text_processor.fill_from_ngrams(self._content)
+
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:698]: keep emply line

File: lab_3_generate_by_ngrams/main.py [hunk 6]
```diff
@@ -708,39 +712,7 @@ def load(self, n_gram_size: int) -> Optional[NGramLanguageModel]:
...
-        n_gram_model.set_n_grams(
-            {
-                ngram: freq / n_grams_cleared_by_size[ngram[:-1]]
-                for ngram, freq in n_grams.items()
-                if len(ngram) == n_gram_size
-            }
-        )
-
-        return n_gram_model
+
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:715]: .

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_generate_by_ngrams/main.py [hunk 7]
```diff
@@ -749,7 +741,7 @@ def get_text_processor(self) -> TextProcessor:  # type: ignore[empty-body]
         Returns:
             TextProcessor: processor created for the current JSON file.
         """
-        return self._text_processor
+
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:744]: remove

File: lab_3_generate_by_ngrams/main.py [hunk 8]
```diff
@@ -773,8 +765,7 @@ def __init__(
             language_models (tuple[NGramLanguageModel]): Language models to use for text generation
             text_processor (TextProcessor): A TextProcessor instance to handle text processing
         """
-        self._language_models = {model.get_n_gram_size(): model for model in language_models}
-        self._text_processor = text_processor
+
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:768]: remove rmpty line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Comment by vitazaytseva** [comment]
Well done! Good job!

**Pull request closed without merging**